src/runtime/policy/ingress-validation.ts

```diff
@@ -226,6 +226,91 @@ export class IngressValidation {
       }
     }
   }
+
+  // Return a new type where all the top-level schemas have been replaced
+  // with the max read schema according to the given `policies`. Returns
+  // null if the given type cannot be handled.
+  //
+  // Suppose that we have types `Foo {a, b, c, d, e, f}` and `Bar {w, x, y, z}`.
+  // Policy is as follows:
+  //   policy MyPolicy {
+  //     from Foo access {a, c, e}
+  //     from Bar access {w, z}
+  //   }
+  //
+  // For various types, the return value of this method is as follows:
+  //
+  //    Foo -> Foo {a, c, e}
+  //   [Foo] -> [Foo {a, c, e}]
+  //   [&Foo] -> [&Foo {a, c, e}]
+  //   (Foo, Bar) -> (Foo {a, c, e}, Bar {w, z})
+  //   List<inline Foo> -> List<inline Foo {a, c, e}>
+  //   Blah -> null
+  //   (Foo, Blah) -> null
+  //   ...
+  public getMaxReadType(type: Type): Type|null {
+    switch (type.tag) {
+      case 'Entity': {
```

Comment:
you could use `EntityType.tag` here instread of hardcoded string

Reply:
This fails because `tag` is property of an instance and not of the type. Did I misunderstand your suggestion?